lib/lambda_ethereum_consensus/p2p/block_downloader.ex

```diff
@@ -44,26 +44,16 @@ defmodule LambdaEthereumConsensus.P2P.BlockDownloader do
     # This should never fail
     {:ok, compressed_payload} = encoded_payload |> Snappy.compress()
 
-    with {:ok, stream} <- Libp2p.host_new_stream(host, peer_id, @blocks_by_range_protocol_id),
-         :ok <- Libp2p.stream_write(stream, size_header <> compressed_payload),
-         :ok <- Libp2p.stream_close_write(stream),
-         {:ok, chunk} <- read_response(stream),
-         {:ok, block} <- decode_response(chunk) do
-      {:ok, block}
-    else
-      {:error, reason} ->
-        if retries > 0 do
-          Logger.debug("Retrying request for block with slot #{slot}")
-          request_block_by_slot(slot, host, retries - 1)
-        else
-          {:error, reason}
-        end
-    end
+    protocol = @blocks_by_range_protocol_id
+    message = size_header <> compressed_payload
+    error_msg = "Retrying request for block with slot #{slot}"
+
+    retrying_request(peer_id, protocol, message, error_msg, retries)
```

Comment:
this seems out of scope of this PR and will conflict with https://github.com/lambdaclass/lambda_ethereum_consensus/pull/341

Reply:
I'll undo the refactor then